### PR TITLE
Properly handle lambdas in relevant domain

### PR DESCRIPTION
**Pull request by ajreynol**
Fix from HO-SMT CADE 2019 branch.

### CODE REVIEW

File: src/theory/quantifiers/relevant_domain.cpp [hunk 1]
```diff
@@ -169,7 +169,9 @@ void RelevantDomain::computeRelevantDomain( Node q, Node n, bool hasPol, bool po
         computeRelevantDomainOpCh( rf, n[i] );
       }
     }
-    if( n[i].getKind()!=FORALL ){
+    // do not recurse under nested quantifiers/lambdas
+    if (n[i].getKind() != FORALL && n[i].getKind() != LAMBDA)
```
Comment (HanielB) [inline src/theory/quantifiers/relevant_domain.cpp]: Why don't we add a general utility function `isBinder` that is called in different places of the code whenever you should have such a test?

Comment (ajreynol) [reply src/theory/quantifiers/relevant_domain.cpp]: Oh, strange, we aren't have it: Node::isClosure.  I'll change to use this.

Comment (HanielB) [reply src/theory/quantifiers/relevant_domain.cpp]: Funny this is not used anywhere. BTW shouldn't `isClosure` also include choice and lets?

Comment (ajreynol) [reply src/theory/quantifiers/relevant_domain.cpp]: There is no such thing as a let. I've already updated it to include choice.

**Pull request merged**